pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -158,6 +162,56 @@ func (*RevisionSpec) applyProbes(container *corev1.Container) {
 	}
 }
 
+//
+func (rs *RevisionSpec) defaultSecurityContext(psc *corev1.PodSecurityContext, container *corev1.Container, cfg *config.Config) {
+	if cfg.Features.SecurePodDefaults != config.Enabled {
+		return
+	}
+
+	if psc == nil {
+		psc = &corev1.PodSecurityContext{}
+	}
+
+	updatedSC := container.SecurityContext
+
+	if updatedSC == nil {
+		updatedSC = &corev1.SecurityContext{}
+	}
+
+	if updatedSC.AllowPrivilegeEscalation == nil {
+		updatedSC.AllowPrivilegeEscalation = ptr.Bool(false)
+	}
+	if psc.SeccompProfile == nil || psc.SeccompProfile.Type == "" {
+		if updatedSC.SeccompProfile == nil {
+			updatedSC.SeccompProfile = &corev1.SeccompProfile{}
+		}
+		if updatedSC.SeccompProfile.Type == "" {
+			updatedSC.SeccompProfile.Type = corev1.SeccompProfileTypeRuntimeDefault
+		}
+	}
+	if updatedSC.Capabilities == nil {
+		updatedSC.Capabilities = &corev1.Capabilities{}
+	}
+	if updatedSC.Capabilities.Drop == nil {
+		updatedSC.Capabilities.Drop = []corev1.Capability{"ALL"}
```

Comment:
If `updatedSC.AllowPrivilegeEscalation` is nil and we set it to false and later on we have `updatedSC.Capabilities.Drop != nil` with `CAP_SYS_ADMIN` present  would that be a problem (https://github.com/kubernetes/website/issues/30104)?

Reply:
Per that note, I don't think we allow `privileged` to be set, so I don't think we get into the forcing issue.

And yes, the idea is that:

1. The flag defaults to false or can be flipped to false, so users can avoid things altogether for a while.

2. The flag only causes the _default_ values to be filled in, so there are ways to explicitly request the old behavior if needed.

Reply:
Don't we allow updatedSC.AllowPrivilegeEscalation set via https://github.com/knative/serving/pull/13395?

Reply:
> Per that note, I don't think we allow `privileged` to be set, so I don't think we get into the forcing issue.

That's right, `SecurityContext.privileged` isn't allowed, though because a _pod's_ security context is behind a feature gate, it fails silently vs. getting caught in the validation as in a _container's_ security context.

> with CAP_SYS_ADMIN present, would that be a problem or K8s 

Since adding capabilities requires enabling a feature gate, we could put extra warnings in the release notes / documentation in for anyone who is flipping that gate on to further mitigate any risk.